src/test/java/io/jenkins/pluginhealth/scoring/probes/ProbeEngineTest.java

```diff
@@ -92,4 +95,40 @@ public void shouldNotSaveErrors() {
 
         verify(plugin, never()).addDetails(any());
     }
+
+    @Test
+    public void shouldBeAbleToGetPreviousContextResultInExecution() throws Exception {
+        final Plugin plugin = spy(Plugin.class);
```

Comment:
Are you using `spy()` on the `Plugin` class because then you won't have to stub the `addDetails()` (and all the other methods inside this class) method and just let it use its concrete implementation?

Reply:
yes.